semgrep-core/src/synthesizing/Range_to_AST.ml

```diff
@@ -90,16 +97,43 @@ let any_at_range r1 ast =
     match expr_at_range r1 ast with None -> None | Some e -> Some (G.E e)
   with Found a -> Some a
 
-let rec many_at_range r1 ast : AST_generic.any list =
-  (* Recurse if there are more tokens in the target. *)
-  let rec_if_more found =
-    match range_of_ast found with
-    (* Failure should never happen? *)
-    | None -> failwith "Could not find token range for target."
-    | Some tokenrange ->
-        let next_start = tokenrange.end_ + 1 in
-        let last = r1.end_ in
-        if next_start > last then [ found ]
-        else found :: many_at_range { start = next_start; end_ = last } ast
+let any_to_stmt (any : G.any) : G.stmt =
+  match any with
+  | G.S s -> s
+  | _ ->
+      failwith
+        (spf "Attempted to convert non-statement to statement %s"
+           (any_to_str any))
+
+let join_anys (anys : AST_generic.any list) : AST_generic.any option =
+  match anys with
+  | [] -> None
+  | [ xs ] -> Some xs
+  | G.S _ :: _ -> Some (G.Ss (List.map any_to_stmt anys))
+  | _ ->
+      failwith
+        "Unable to handle ranges that contain multiple expressions. Range must \
+         contain a single expression or one-or-more statements."
+
+let split_any (any : G.any) : G.any list =
```

Comment:
Perhaps `split_stmts` would be a better name for this, since it's unclear what the node is being split on otherwise